### PR TITLE
KAA-1398: Create a Java code style page

### PR DESCRIPTION
**Pull request by Acarus**

### CODE REVIEW

File: doc/Customization-guide/Code-style/Java/index.md [hunk 1]
```diff
@@ -4,4 +4,39 @@ title: Java
...
+This guide describes tools which we use to check and format a code.
+
+# IntelliJ IDEA plugin
+[Checkstyle](https://github.com/jshiell/checkstyle-idea) plugin installation:
+1. Go to *File>Settings>IDE Settings>plugins*.
+2. Click on the *Browse repositories* button.
+3. Type *Checkstyle-idea* and install the plugin.
+4. Download the *[configuration for the Google coding conventions](https://github.com/checkstyle/checkstyle/blob/master/src/main/resources/google_checks.xml)*.
+5. Go to *File>Settings>IDE Settings>CheckStyle*.
+6. Add the configuration file (*google_checks.xml*).
```
Comment (rasendubi) [inline doc/Customization-guide/Code-style/Java/index.md]: Name of the file should be in backticks.

File: doc/Customization-guide/Code-style/Java/index.md [hunk 2]
```diff
@@ -4,4 +4,39 @@ title: Java
...
+Use the next [tutorial](http://eclipse-cs.sourceforge.net/#!/install) to install the checkstyle plugin.
+
+# Maven plugin
+We use [check style plugin](https://maven.apache.org/plugins/maven-checkstyle-plugin/usage.html) with [configuration for the Google coding conventions](https://github.com/checkstyle/checkstyle/blob/master/src/main/resources/google_checks.xml).
+
+# Automatic code formatter
+We use IntelliJ IDEA to reformat Java source code to comply with Google Java Style.<br> 
+Installing the coding style settings:
+1. Download [intellij-java-google-style.xml](https://github.com/google/styleguide/blob/gh-pages/intellij-java-google-style.xml).
+2. Go to File>Settings>Editor>Code Style.
```
Comment (rasendubi) [inline doc/Customization-guide/Code-style/Java/index.md]: put italics

File: doc/Customization-guide/Code-style/Java/index.md [hunk 3]
```diff
@@ -4,4 +4,39 @@ title: Java
...
+3. Press *Manage* button and import *intellij-java-google-style.xml*.
```
Comment (rasendubi) [inline doc/Customization-guide/Code-style/Java/index.md]: backticks

**Comment by rasendubi** [comment]
It's better to avoid `<br>`s.

**Comment by rasendubi** [comment]
LGTM
cc @ashvayka @ikulikov

File: doc/Customization-guide/Code-style/Java/index.md [hunk 4]
```diff
@@ -4,4 +4,39 @@ title: Java
 permalink: /:path/
 sort_idx: 10
 ---
-Specific per-language code style guidelines. Check style configuration files for Eclipse and IDEA. Style check also must be automated in the build process (after code styles are defined here).
+
+* TOC
+{:toc}
+
+We follow [Google Java Style](https://google.github.io/styleguide/javaguide.html) in Kaa Java SDK.  
+This guide describes tools which we use to check and format a code.
```
Comment (ikulikov) [inline doc/Customization-guide/Code-style/Java/index.md]: This guide describes tools used to check style and format Java source code.

File: doc/Customization-guide/Code-style/Java/index.md [hunk 5]
```diff
@@ -4,4 +4,39 @@ title: Java
...
+Plugin configuration:
+1. Go to *Analyze>Inspect code*.
+2. Create a new profile with the checkstyle inspector.
+
+Plugin usage:
+1. Go to *Analyze>Inspect code*.
+2. Select an inspection scope and an inspection profile.
+
+# Eclipse plugin
+Use the next [tutorial](http://eclipse-cs.sourceforge.net/#!/install) to install the checkstyle plugin.
```
Comment (ikulikov) [inline doc/Customization-guide/Code-style/Java/index.md]: Follow the next ...

File: doc/Customization-guide/Code-style/Java/index.md [hunk 6]
```diff
@@ -4,4 +4,39 @@ title: Java
...
+
+# Maven plugin
+We use [check style plugin](https://maven.apache.org/plugins/maven-checkstyle-plugin/usage.html) with [configuration for the Google coding conventions](https://github.com/checkstyle/checkstyle/blob/master/src/main/resources/google_checks.xml).
```
Comment (ikulikov) [inline doc/Customization-guide/Code-style/Java/index.md]: In order to perform continuous java code style check the [check style maven plugin](https://maven.apache.org/plugins/maven-checkstyle-plugin/usage.html) is used with [configuration for the Google coding conventions](https://github.com/checkstyle/checkstyle/blob/master/src/main/resources/google_checks.xml).

File: doc/Customization-guide/Code-style/Java/index.md [hunk 7]
```diff
@@ -4,4 +4,39 @@ title: Java
...
+
+# Automatic code formatter
+We use IntelliJ IDEA to reformat Java source code to comply with Google Java Style.
```
Comment (ikulikov) [inline doc/Customization-guide/Code-style/Java/index.md]: In order to automate Java code formatting it is preferred to use IntelliJ IDEA with appropriate Google Java Style settings.

File: doc/Customization-guide/Code-style/Java/index.md [hunk 8]
```diff
@@ -4,4 +4,39 @@ title: Java
...
+
+# Eclipse plugin
+Use the next [tutorial](http://eclipse-cs.sourceforge.net/#!/install) to install the checkstyle plugin.
+
+# Maven plugin
+We use [check style plugin](https://maven.apache.org/plugins/maven-checkstyle-plugin/usage.html) with [configuration for the Google coding conventions](https://github.com/checkstyle/checkstyle/blob/master/src/main/resources/google_checks.xml).
+
+# Automatic code formatter
+We use IntelliJ IDEA to reformat Java source code to comply with Google Java Style.   
+Installing the coding style settings:
```
Comment (ikulikov) [inline doc/Customization-guide/Code-style/Java/index.md]: ... the code style ...

File: doc/Customization-guide/Code-style/Java/index.md [hunk 9]
```diff
@@ -4,4 +4,39 @@ title: Java
...
+1. Download [intellij java google style](https://github.com/google/styleguide/blob/gh-pages/intellij-java-google-style.xml).
```
Comment (ikulikov) [inline doc/Customization-guide/Code-style/Java/index.md]: Download the [IntelliJ Java Google Style]

File: doc/Customization-guide/Code-style/Java/index.md [hunk 10]
```diff
@@ -4,4 +4,39 @@ title: Java
 permalink: /:path/
 sort_idx: 10
 ---
-Specific per-language code style guidelines. Check style configuration files for Eclipse and IDEA. Style check also must be automated in the build process (after code styles are defined here).
+
+* TOC
+{:toc}
+
+We follow [Google Java Style](https://google.github.io/styleguide/javaguide.html) in Kaa Java SDK.
```
Comment (ikulikov) [inline doc/Customization-guide/Code-style/Java/index.md]: Whole Java source code of Kaa project must conform to the [Google Java Style](https://google.github.io/styleguide/javaguide.html).

**Pull request merged**